internal/agent/agent.go

```diff
@@ -64,7 +76,7 @@ type Agent struct {
 //
 // Returns:
 //   - A pointer to a newly created and initialized Agent instance.
-func NewAgent(address string, contentType string, pollInterval time.Duration, reportInterval time.Duration, secretKey []byte, rateLimit int) *Agent {
+func NewAgent(address string, contentType string, pollInterval time.Duration, reportInterval time.Duration, secretKey []byte, rateLimit int, cryptoKeyDir string) *Agent {
```

Comment:
```suggestion
func NewAgent(address string, contentType string, pollInterval time.Duration, reportInterval time.Duration, secretKey []byte, rateLimit int, cryptoKeyDir string) (*Agent, error) {
```
и в main уже дергать Fatal